src/main/java/com/robertx22/mine_and_slash/mechanics/thresholds/datapack/SpendThresholdDef.java

```diff
@@ -0,0 +1,95 @@
+package com.robertx22.mine_and_slash.mechanics.thresholds.datapack;
+
+import com.google.gson.annotations.SerializedName;
+import com.robertx22.mine_and_slash.mechanics.thresholds.DataDrivenSpendThresholdSpec;
+import com.robertx22.mine_and_slash.mechanics.thresholds.SpendThresholdSpec;
+import com.robertx22.mine_and_slash.saveclasses.unit.ResourceType;
+import net.minecraft.server.level.ServerPlayer;
+
+import java.util.*;
+
+public class SpendThresholdDef {
+    public String key;
+    public String resource = "energy";
```

Comment:
It's up to your vision of this, but maybe there should be no default value
```suggestion
    public String resource = "";
```